eudemo/eudemo/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+        x_ib: float,
+        x_ob: float,
+    ):
+        super().__init__(params, build_config)
+        self.x_ib = x_ib
+        self.x_ob = x_ob
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class EquatorialPortBuilder(Builder):
+    """
+    Equatorial Port Builder
+    """
+
+    NAME_XY = "Equatorial Ports xy"
+    NAME_XYZ = "Equatorial Ports xyz"
+    KOZ = "koz"
+    param_cls: Type[EquatorialPortBuilderParams] = EquatorialPortBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortBuilderParams],
+        build_config: Union[Dict, None],
+        xz_profile: BluemiraFace,
+        y_width: float,
+        x_offsets: list,
+        castellation_offsets: list,
+    ):
+        super().__init__(params, build_config)
+        self.xz_profile = xz_profile
+        self.y_width = y_width
+        self.x_off = x_offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the equatorial port component.
+        """
+        # TODO: Implement corner radii
+        self.r_rad = self.params.ep_r_corner.value
+        self.n_ep = self.params.n_ep.value
+
+        # Set-up coordinates for geometry building
+        xz_plane = BluemiraPlane.from_3_points([0, 0, 0], [1, 0, 0], [1, 0, 1])
+        intersections = slice_shape(self.xz_profile.boundary[0], xz_plane)
+        self.x_ib = min(intersections[:, 0])
+        self.x_ob = max(intersections[:, 0])
+        self.z_lo = min(intersections[:, 2])
+        self.z_hi = max(intersections[:, 2])
+
+        x_points, y_points, z_points, xyz_coords = self.build_coordinates(
+            self.x_off, self.cst
+        )
+
+        return self.component_tree(
+            xz=[self.build_xz(x_points, z_points)],
+            xy=self.build_xy(x_points, y_points),
+            xyz=self.build_xyz(xyz_coords, n_ep=10),
+        )
+
+    def build_xz(self, x: list, z: list) -> PhysicalComponent:
+        """
+        Build the xz components of the equatorial port
+        """
+        xz_profile = BluemiraWire(make_polygon({"x": x, "y": 0, "z": z}, closed=True))
+        body = PhysicalComponent(self.KOZ, xz_profile)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+        return body
+
+    def build_xy(self, x: list, y: list, n_ep: int = 10) -> List[PhysicalComponent]:
+        """
+        Build the xy components of the equatorial port
+        """
+        ep_xy_profile = BluemiraWire(make_polygon({"x": x, "y": y, "z": 0}, closed=True))
```

Comment:
It looks like both the xy and xz views dont follow the same shape as the xyz, I added `output.show_cad()` in your test_ep_builder (and only created one xyz shape). This is shown when pytest is run with the `--plotting-on` flag:

![Screenshot from 2023-03-14 14-18-10](https://user-images.githubusercontent.com/81617086/225030154-b65efa23-4374-470e-807b-6fec988cc952.png)

Reply:
Resolved with a bit of zip() help